titus-server-master/src/main/java/com/netflix/titus/master/service/management/internal/DefaultApplicationSlaManagementService.java

```diff
@@ -52,6 +52,12 @@ public DefaultApplicationSlaManagementService(CapacityMonitoringService capacity
                 .toList().toBlocking().firstOrDefault(Collections.emptyList());
     }
 
+    @Override
+    public Collection<ApplicationSLA> getApplicationSLAsForScheduler(String schedulerName) {
+        return storage.findBySchedulerName(schedulerName).onErrorReturn(t -> null)
```

Comment:
Does `.onErrorReturn(t -> null)` emit null value or  nothing at all?

Reply:
It emits a null value that would appear in the return collection. This behavior is borrowed from the other get methods in this service. Should we do anything differently here and possibly at other places?

Reply:
Yes, this is incorrect. Could you also fix the other method? What happens if a list of null is returned?

Reply:
Changed `onErrorReturn()` to `doOnError()` and added error logging. Please review.